JS/module/Auth/validation.js

```diff
@@ -0,0 +1,62 @@
+import { inputEmail, errorEmail, inputPassword, errorPassword, inputPassword2, errorPassword2} from "./variables.js";
+import { removeErrorStyle, errorStyle, EMAIL_EMPTY,EMAIL_NOT_VALID,EMAIL_OCCUPIED,PASSWORD_EMPTY,PASSWORD_NOT_SAME,PASSWORD_NOT_VALID } from "./errorStyle.js";
+import { checkFormEmpty, checkEmailValid, checkPasswordValid,checkStringSame,checkEmailAvailable} from "./conditions.js";
+
+export function emailValidationSignIn(){
+  const email = inputEmail.value;
+  if (!checkFormEmpty(email))
+    return errorStyle(inputEmail, errorEmail,EMAIL_EMPTY)
+  else if (!checkEmailValid(email)) {
+    return errorStyle(inputEmail, errorEmail, EMAIL_NOT_VALID)
+  } else {
+    return true
+  }
+}
+
+export function passwordValidationSignIn(){
+  const password = inputPassword.value;
+  if (!checkFormEmpty(password)){
+  errorStyle(inputPassword, errorPassword, PASSWORD_EMPTY)
+  return false
+  }
+  return true
+}
+
+export async function emailValidationSignUp() {
+  const email = inputEmail.value;
+  if (!checkFormEmpty(email)) {
+    return errorStyle(inputEmail,errorEmail,EMAIL_EMPTY);
+  } else if (!checkEmailValid(email)) {
+    return errorStyle(inputEmail,errorEmail,EMAIL_NOT_VALID);
+  } else if (!await checkEmailAvailable()) {
+    return errorStyle(inputEmail,errorEmail,EMAIL_OCCUPIED);
+  } else {
+    removeErrorStyle(inputEmail, errorEmail);
+    return true;
+  }
```

Comment:
저 이번에 리팩토링할 때 얼리리턴 패턴을 해보라고 하셔서 해봤는데 이런 else if문들 많을 때 쓰니까 확실히 깔끔해지더라고여
한번 해 보시는 걸 추천드립니다! 신세계!

Reply:
도전! 해보겠습니다 🧐

Reply:
```suggestion
export async function emailValidationSignUp() {
  const email = inputEmail.value;
  if (!checkFormEmpty(email)) return errorStyle(inputEmail,errorEmail,EMAIL_EMPTY);
  if (!checkEmailValid(email)) return errorStyle(inputEmail,errorEmail,EMAIL_NOT_VALID);
  if (!await checkEmailAvailable()) return errorStyle(inputEmail,errorEmail,EMAIL_OCCUPIED);
  removeErrorStyle(inputEmail, errorEmail);
  return true;
```
저라면 얼리 리턴 이렇게 적용할 것 같아용
- 이상 suggestion에 맛들린 사람입니당 ..ㅋㅋㅋㅋ

Reply:
저 이렇게 리뷰 꼼꼼하게 봐주셔서 울어요 광광